go/tools/gazelle/rules/generator.go

```diff
@@ -105,11 +105,13 @@ func (g *generator) Generate(rel string, pkg *build.Package) ([]*bzl.Rule, error
 	}
 
 	library := defaultLibName
-	r, err := g.generateLib(rel, library, pkg, cgoLibrary)
+	libRule, err := g.generateLib(rel, library, pkg, cgoLibrary)
 	if err != nil {
 		return nil, err
 	}
-	rules = append(rules, r)
+	if libRule != nil {
+		rules = append(rules, libRule)
```

Comment:
IIUC,
when `len(pkg.CgoFiles) > 0 && len(pkg.GoFiles) == 0 && len(pkg.SFiles) == 0` `generateLib` returns `nil`
In this case, and if `pkg.IsCommand()` is `true`. the generated `go_binary` target depends on a `go_library` rule named `go_default_library` but no such target is generated.